creation/create_cvmfsexec_distros.sh

```diff
@@ -17,86 +17,113 @@ CVMFS_SOURCES=osg:egi:default
 SUPPORTED_TYPES=rhel7-x86_64:rhel8-x86_64:suse15-x86_64
-cvmfsexec_temp=$(mktemp -d -t cvmfsexec.XXX)
-cvmfsexec_base=$cvmfsexec_temp/cvmfsexec
-cvmfsexec_latest=$cvmfsexec_temp/latest
-cvmfsexec_distros=$cvmfsexec_temp/distros
-work_dir=$(grep -m 1 "submit" /etc/gwms-factory/glideinWMS.xml | awk -F"\"" '{print $6}')
-cvmfsexec_tarballs=$work_dir/cvmfsexec/tarballs
-
-if [[ -d $cvmfsexec_temp ]]; then
-#    rm -rf $cvmfsexec_pkg
-    if [[ -d $cvmfsexec_base ]]; then
-        curr_ver=`$cvmfsexec_base/cvmfsexec -v`
-        git clone https://www.github.com/cvmfs/cvmfsexec.git $cvmfsexec_latest &> /dev/null
-        latest_ver=`$cvmfsexec_latest/cvmfsexec -v`
-        if [[ $curr_ver != $latest_ver ]]; then
-            # if current version and latest version are different, use the latest
-            echo "Current version of cvmfsexec: $curr_ver"
-            echo "Found newer version of cvmfsexec..."
-            rm -rf $cvmfsexec_base
-            mv $cvmfsexec_latest $cvmfsexec_base
-            echo "Latest version of cvmfsexec: `$cvmfsexec_base/cvmfsexec -v`"
-            echo "Using cvmfsexec version: `$cvmfsexec_base/cvmfsexec -v`"
-        else
-            # if current version and latest version are the same
-            echo "Current version and latest version of cvmfsexec are identical!"
-            echo "cvmfsexec version: `$cvmfsexec_base/cvmfsexec -v`"
-            rm -rf $cvmfsexec_latest
-            exit 0
-        fi
-    else
-        # $cvmfsexec_base does not exist
-        git clone https://www.github.com/cvmfs/cvmfsexec.git $cvmfsexec_base &> /dev/null
-        echo "cvmfsexec version: `$cvmfsexec_base/cvmfsexec -v`"
-    fi
-else
-    # $cvmfsexec_temp does not exist
-    git clone https://www.github.com/cvmfs/cvmfsexec.git $cvmfsexec_base &> /dev/null
-    echo "cvmfsexec version: `$cvmfsexec_base/cvmfsexec -v`"
+factory_config_file="/etc/gwms-factory/glideinWMS.xml"
 
+sys_tmp=$(echo $TMPDIR)
+if [[ -z "$sys_tmp" ]]; then
+    sys_tmp=/tmp
 fi
 
-if [[ ! -d $cvmfsexec_distros ]]; then
-    mkdir -p $cvmfsexec_distros
+# first, checking in the work-dir location for the current version of cvmfsexec
+work_dir=$(grep -m 1 "submit" "$factory_config_file" | sed 's/.*base_dir="\([^"]*\)".*/\1/')
+# protect aginst non-existence of cvmfsexec directory; fresh install of GWMS with first run of factory upgrade
+if [[ -d "$work_dir/cvmfsexec" && -d "$work_dir/cvmfsexec/tarballs" ]]; then
+    cvmfsexec_tarballs=$work_dir/cvmfsexec/tarballs
+    if [[ -f "$cvmfsexec_tarballs/.cvmfsexec_version" ]]; then
+        curr_ver=$(cat "$cvmfsexec_tarballs"/.cvmfsexec_version)
+        echo "Current version found: $curr_ver"
+    fi
+else
+    # if the cvmfsexec directory does not exist
+    mkdir -p $work_dir/cvmfsexec
+    # create a directory named tarballs under cvmfsexec directory
+    cvmfsexec_tarballs=$(mkdir -p $work_dir/cvmfsexec/tarballs -v | awk -F" " '{print $4}' | sed 's/^.//; s/.$//' )
```

Comment:
Thank you for the notes on best practices.

> * be as specific as possible. If you have to remove a quote, remove a quote, not the first character whatever it is (`sed "s/^'//; s/'$//"` -may need to quote, not sure-), even better bash builtins avoid the invocation of a subshell:
> 
> ```
> cvmfsexec_tarballs="${cvmfsexec_tarballs#'}"
> cvmfsexec_tarballs="${cvmfsexec_tarballs%'}"
> ```
I first tried with the sed pattern that you suggested, which is to remove the quote character rather than the first or last character. However, the kind of quote character in the output of `mkdir -v` was not being matched with the pattern as it was a different style of quotes. This was the sole reason I resorted to removing the first and last character.


> Then I do not understand the reason of the pipe, it will return "$work_dir/cvmfsexec/tarballs" if all is OK

I used the pipe to be able to store the location of the directory that was newly created instead of initializing with `$work-dir/cvmfsexec/tarballs`.

I like the suggestion of including a check for failure of directory creation. I've incorporated the same in one of the new commits.